packages/web_ui/src/components/SlaveViewPage.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useContext, useState } from "react";
+import { Descriptions, PageHeader, Spin, Typography } from "antd";
+import { useParams } from "react-router-dom";
+
+import InstanceList from "./InstanceList";
+import LogConsole from "./LogConsole";
+import { useAccount } from "../model/account";
+import { useInstanceList } from "../model/instance";
+import { useSlave } from "../model/slave";
+import PageLayout from "./PageLayout";
+
+const { Title } = Typography;
+
+
+export default function SlaveViewPage(props) {
+	let params = useParams();
+	let slaveId = Number(params.id);
+	let account = useAccount();
+	let [instanceList] = useInstanceList();
+	let [slave] = useSlave(slaveId);
+
+	instanceList = instanceList.filter(instance => instance["assigned_slave"] === slaveId);
+
+	let nav = [{ name: "Slaves", path: "/slaves" }, { name: slave.name || slaveId }];
+	if (slave.loading) {
+		return <PageLayout nav={nav}><Spin size="large" /></PageLayout>;
+	}
+
+	if (slave.missing) {
+		return <PageLayout nav={nav}>
+			<PageHeader
+				className="site-page-header"
+				title={slaveId}
+			/>
+			<p>Slave with id {slaveId} was not found on the master server.</p>
+		</PageLayout>;
+	}
+
+	return <PageLayout nav={nav}>
+		<PageHeader
+			className="site-page-header"
+			title={slave.name || slaveId}
+		/>
+
+		<Descriptions bordered size="small" column={{ xs: 1, sm: 2, xl: 4 }}>
+			<Descriptions.Item label="Name">{slave["name"]}</Descriptions.Item>
```

Comment:
Is there a reason why you use bracket syntax here instead of . syntax?

Reply:
It's a hint that it's a plain object, like deserialized JSON.

Reply:
Ah, I see. I will try to keep to the same pattern then.